theforeman.org/pipelines/test/smart-proxy.groovy

```diff
@@ -67,7 +67,12 @@ pipeline {
                 axes {
                     axis {
                         name 'ruby'
-                        values '2.7'
+                        values '3.1'
```

Comment:
We want to test a matrix, namely 2.7, 3.0 and 3.1. This is because we still want to make sure Ruby 2.7 continues working. There may also be differences between 3.0 and 3.1 so we want to verify both.
```suggestion
                        values '2.7', '3.0', '3.1'
```

Reply:
I think this will result in running release branches against Ruby 3 versions as well. We can handle this by adapting the logic from Foreman's tests:

 * Add a when condition: https://github.com/theforeman/jenkins-jobs/blob/master/theforeman.org/pipelines/test/foreman/unit.groovy#L23-L27
 * https://github.com/theforeman/jenkins-jobs/blob/master/theforeman.org/pipelines/test/foreman/unit.groovy#L2
 * Update https://github.com/theforeman/jenkins-jobs/blob/91b6fcf3cf7b11b15964eba9228d05431d9b741a/theforeman.org/pipelines/test/foreman/vars.groovy#L32

Reply:
We had an offline discussion, but I recommended to just proceed with the simple version. In  https://github.com/theforeman/smart-proxy/commit/a921761d7f8ba5f2aadb793872cc17b5ba1b062f (https://projects.theforeman.org/issues/35510) the tests started to pass. I now see that was only in 3.5.0 and we still support 3.4 now.

Though I do think it's easier if we add `webrick` to Gemfile in 3.4-stable. That way we do pull it in for dev setups, but don't make any packaging change. Then I'd expect it to just pass and the CI code is MUCH easier.

Reply:
That's fine, it's likely this will have to be re-visited on the backend where we start dropping Ruby 2.7 support but our release branches still need it.

Reply:
Yes, but I'm fine with making that a "future us"-problem ;)